tracer/src/Datadog.Trace/Ci/Agent/Payloads/EvpPayload.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="EvpPayload.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.Ci.Agent.Payloads
+{
+    /// <summary>
+    /// Event-platform payload
+    /// </summary>
+    internal abstract class EvpPayload
+    {
+        private Uri _url;
+
+        /// <summary>
+        /// Gets Event-Platform subdomain/track
+        /// </summary>
+        public abstract string EvpSubdomain { get; }
+
+        /// <summary>
+        /// Gets Event-Platform path
+        /// </summary>
+        public abstract string EvpPath { get; }
+
+        /// <summary>
+        /// Gets or sets the Payload url
+        /// </summary>
+        public Uri Url
+        {
+            get
+            {
+                if (_url is { } url)
+                {
+                    return url;
+                }
+
+                UriBuilder builder;
+                if (CIVisibility.Settings.Agentless)
+                {
+                    var agentlessUrl = CIVisibility.Settings.AgentlessUrl;
```

Comment:
I still dislike all these statics 😅

Reply:
I did a refactor to remove them in [1e4169d](https://github.com/DataDog/dd-trace-dotnet/pull/3032/commits/1e4169d0415456267bdee4a0ca2ffe629a9a8e1b)